x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/operator/DefaultOperatorPrivilegesTests.java

```diff
@@ -265,4 +268,21 @@ public void testNoOpServiceMaybeInterceptRequest() {
         NOOP_OPERATOR_PRIVILEGES_SERVICE.maybeInterceptRequest(threadContext, mock(TransportRequest.class));
     }
 
+    public void testCheckRest() {
+        final Settings settings = Settings.builder().put("xpack.security.operator_privileges.enabled", true).build();
+        when(xPackLicenseState.isAllowed(Security.OPERATOR_PRIVILEGES_FEATURE)).thenReturn(true);
+        RestHandler restHandler = mock(RestHandler.class);
+        RestRequest restRequest = mock(RestRequest.class);
+        RestChannel restChannel = mock(RestChannel.class);
+        ThreadContext threadContext = new ThreadContext(settings);
+
+        // not an operator
+        when(operatorOnlyRegistry.checkRest(restHandler, restRequest, restChannel)).thenReturn(() -> "violation!");
+        assertFalse(operatorPrivilegesService.checkRest(restHandler, restRequest, restChannel, threadContext));
+
+        // is an operator
+        threadContext.putHeader(AuthenticationField.PRIVILEGE_CATEGORY_KEY, AuthenticationField.PRIVILEGE_CATEGORY_VALUE_OPERATOR);
+        when(operatorOnlyRegistry.checkRest(restHandler, restRequest, restChannel)).thenReturn(null);
```

Comment:
Nit: Since we essentially skips check if the uers is an operator, it does not matter whether `checkRest` returns null or not. Technically, it will actually return violation if it is invoked. So maybe we could just randomize it here and comment to say it does not matter because of the order of execution.